be/src/storage/schema_change.cpp

```diff
@@ -1113,10 +1113,13 @@ Status SchemaChangeWithSorting::processV2(vectorized::TabletReader* reader, Rows
     vectorized::Schema new_schema = std::move(ChunkHelper::convert_schema_to_format_v2(new_tablet->tablet_schema()));
     auto char_field_indexes = std::move(vectorized::ChunkHelper::get_char_field_indexes(new_schema));
 
-    // memtable max buffer size set 80% of memory limit so that it will do _merge() if reach 80%
-    // do finialize() and flush() if reach 90%
-    auto mem_table = std::make_unique<MemTable>(new_tablet->tablet_id(), new_schema, new_rowset_writer,
-                                                _memory_limitation * 0.8, tls_thread_status.mem_tracker());
+    // memtable max buffer size set default 80% of memory limit so that it will do _merge() if reach limit
+    // set max memtable size to 4G since large buffer will make large segment & decrease sort performace
+    // TODO: try to use accurate memory limit
```

Comment:
better add one comment: Currently, MemTable and Segment size cannot exceed 4G, otherwise, it is possible to write dirty data

Reply:
Please reopen the pr